Stack/Opc.Ua.Core/Stack/Transport/ITransportListener.cs

```diff
@@ -32,12 +36,99 @@ public interface ITransportListener : IDisposable
             TransportListenerSettings settings,
-            ITransportListenerCallback callback);
+            ITransportListenerCallback callback
+            );
 
         /// <summary>
         /// Closes the listener and stops accepting connection.
         /// </summary>
         /// <exception cref="ServiceResultException">Thrown if any communication error occurs.</exception>
         void Close();
+
+        /// <summary>
+        /// Updates the application certificate for a listener.
+        /// </summary>
+        void CertificateUpdate(
+            ICertificateValidator validator,
+            X509Certificate2 serverCertificate,
+            X509Certificate2Collection serverCertificateChain);
+
+        /// <summary>
+        /// Raised when a new connection is waiting for a client.
+        /// </summary>
+        event EventHandler<ConnectionWaitingEventArgs> ConnectionWaiting;
+
+        /// <summary>
+        /// Raised when a monitored connection's status changed.
+        /// </summary>
+        event EventHandler<ConnectionStatusEventArgs> ConnectionStatusChanged;
+
+        /// <summary>
+        /// Creates a reverse connection to a client. 
+        /// </summary>
+        void CreateReverseConnection(Uri url, int timeout);
+
+    }
+
+    /// <summary>
+    /// The arguments passed to the ConnectionWaiting event. 
+    /// </summary>
+    public interface ITransportWaitingConnection
+    {
+        /// <remarks/>
+        string ServerUri { get; }
+
+        /// <remarks/>
+        Uri EndpointUrl { get; }
+
+        /// <remarks/>
+        object Handle { get; }
+    }
+
+    /// <summary>
+    /// The arguments passed to the ConnectionWaiting event. 
+    /// </summary>
+    public class ConnectionWaitingEventArgs : EventArgs, ITransportWaitingConnection
+    {
+        protected ConnectionWaitingEventArgs(string serverUrl, Uri endpointUrl)
+        {
+            ServerUri = serverUrl;
+            EndpointUrl = endpointUrl;
+            Accepted = false;
+        }
+
+        /// <remarks/>
+        public string ServerUri { get; private set; }
+
+        /// <remarks/>
+        public Uri EndpointUrl { get; private set; }
+
+        /// <remarks/>
+        public virtual object Handle => null;
+
+        /// <remarks/>
+        public bool Accepted { get; set; }
+    }
+
+    /// <summary>
+    /// The arguments passed to the ConnectionStatus event. 
+    /// </summary>
+    public class ConnectionStatusEventArgs : EventArgs
+    {
+        internal ConnectionStatusEventArgs(Uri endpointUrl, ServiceResult channelStatus, bool closed)
+        {
+            EndpointUrl = endpointUrl;
+            ChannelStatus = channelStatus;
+            Closed = closed;
+        }
+
+        /// <remarks/>
+        public Uri EndpointUrl { get; private set; }
+
+        /// <remarks/>
+        public ServiceResult ChannelStatus { get; private set; }
+
+        /// <remarks/>
+        public bool Closed { get; private set; }
     }
```

Comment:
[Quality] interface file's should not contain classes

Reply:
moved to connectionevents.cs

---
In reply to: [432431372](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432431372) [](ancestors = 432431372)